src/components/Checkbox/Checkbox.tsx

```diff
@@ -1,25 +1,39 @@
-import {
-  Indicator,
-  Root as RadixCheckbox,
-  CheckboxProps as RadixCheckboxProps,
-} from "@radix-ui/react-checkbox";
+import { Indicator, Root as RadixCheckbox } from "@radix-ui/react-checkbox";
 import { ReactNode, forwardRef } from "react";
 import { classNames } from "~/utils";
-import { Box } from "../Box";
+import { Box, PropsWithBox } from "../Box";
 import { CheckedIcon } from "./CheckedIcon";
 import { IndeterminateIcon } from "./IndeterminateIcon";
 import { commonCheckbox, defaultCheckbox, errorCheckbox } from "./Checkbox.css";
 
-export type CheckboxProps = RadixCheckboxProps & {
-  error?: boolean;
+export type CheckboxProps = PropsWithBox<{
   children?: ReactNode;
-};
+  error?: boolean;
+  defaultChecked?: boolean;
+  checked?: CheckedState;
```

Comment:
**question**: what's the difference between those two? Any use case?

Reply:
`defaultChecked` determines state of the checkbox on initial render when it is uncontrolled. These props are pretty much mutually exclusive because using `checked` makes the component controlled.